cmake/template_instantiation.cmake

```diff
@@ -0,0 +1,68 @@
+function(add_instantiation_files source_dir source_file output_files_var)
+    # read full file into variable
+    set(source_path "${source_dir}/${source_file}")
+    file(READ "${source_path}" file_contents)
```

Comment:
if the \r\n is not critical, you can also use `file(STRINGS ...)` which ignores \r

Reply:
It seems like `STRINGS` does not properly escape semicolons, thus we lose them in the output. This is the reason I do this weird escaping step first